asapdiscovery-ml/asapdiscovery/ml/schema.py

```diff
@@ -0,0 +1,934 @@
+            )
+            for split, split_d in target_vals.items()
+        }  # maps split -> target_vals (n_compounds,) ordered by compound_ids_dict
+
+        # Dict to accumulate results
+        stats_dict = {"train": {}, "val": {}, "test": {}}
+
+        # Loop through splits
+        for sp in stats_dict.keys():
+            preds = preds_dict[sp]
+            target_vals = target_vals_dict[sp]
+
+            # Calculate MAE and bootstrapped confidence interval
+            mae = np.abs(target_vals - preds).mean()
+            conf_interval = bootstrap(
+                (target_vals, preds),
+                statistic=lambda target, pred: np.abs(target - pred).mean(),
+                method="basic",
+                confidence_level=0.95,
+                paired=True,
+            ).confidence_interval
+            stats_dict[sp]["mae"] = {
+                "value": mae,
+                "95ci_low": conf_interval.low,
+                "95ci_high": conf_interval.high,
+            }
+
+            # Calculate RMSE and bootstrapped confidence interval
+            rmse = np.sqrt(np.power(target_vals - preds, 2).mean())
+            conf_interval = bootstrap(
+                (target_vals, preds),
+                statistic=lambda target, pred: np.sqrt(
+                    np.power(target - pred, 2).mean()
+                ),
+                method="basic",
+                confidence_level=0.95,
+                paired=True,
+            ).confidence_interval
+            stats_dict[sp]["rmse"] = {
+                "value": rmse,
+                "95ci_low": conf_interval.low,
+                "95ci_high": conf_interval.high,
+            }
+
+            # Calculate Spearman r and bootstrapped confidence interval
+            sp_r = spearmanr(target_vals, preds).statistic
+            conf_interval = bootstrap(
+                (target_vals, preds),
+                statistic=lambda target, pred: spearmanr(target, pred).statistic,
+                method="basic",
+                confidence_level=0.95,
+                paired=True,
+            ).confidence_interval
+            stats_dict[sp]["sp_r"] = {
+                "value": sp_r,
+                "95ci_low": conf_interval.low,
+                "95ci_high": conf_interval.high,
+            }
+
+            # Calculate Kendall's tau and bootstrapped confidence interval
+            tau = kendalltau(target_vals, preds).statistic
+            conf_interval = bootstrap(
+                (target_vals, preds),
+                statistic=lambda target, pred: kendalltau(target, pred).statistic,
+                method="basic",
+                confidence_level=0.95,
+                paired=True,
+            ).confidence_interval
+            stats_dict[sp]["tau"] = {
+                "value": tau,
+                "95ci_low": conf_interval.low,
+                "95ci_high": conf_interval.high,
+            }
+
+        return stats_dict
```

Comment:
This is a lot of code, is there any chance we can drop some? I have no ability to really vouch for / test it so perhaps cutting down volume if possible.